util/chplenv/chpl_3p_hwloc_configs.py

```diff
@@ -14,5 +14,11 @@ def get_uniq_cfg_path():
                             chpl_locale_model.get())
 
 @memoize
-def get_link_args():
-    return third_party_utils.default_get_link_args('hwloc')
+def get_link_args(hwloc):
+    if hwloc == 'hwloc':
+        return third_party_utils.default_get_link_args('hwloc',
+                                                       ucp=get_uniq_cfg_path())
```

Comment:
because otherwise the link args don't get passed the correct directory. 

This file defines a custom get_uniq_cfg_path:
``` python
def get_uniq_cfg_path():
    return '{0}-{1}'.format(third_party_utils.default_uniq_cfg_path(),
                            chpl_locale_model.get())
```

and then in third_party_utils.py we have:
``` python
def default_uniq_cfg_path():
    return '{0}-{1}-{2}'.format(chpl_platform.get('target'),
                                chpl_compiler.get('target'),
                                chpl_arch.get('target', map_to_compiler=True,
                                              get_lcd=using_chapel_module()))
```

which is not the same as chpl_3p_hwloc_configs.get_uniq_cfg_path. Then the default_get_link_args calls the above if ucp is not set:

``` python
def default_get_link_args(pkg, ucp='', libs=[]):
    if ucp == '':
        ucp = default_uniq_cfg_path()
```

So, any chpl_3p_*configs.py needs to pass ucp to default_get_link_args if it has a custom get_uniq_cfg_path.